slatedb/src/dispatcher.rs

```diff
@@ -210,8 +217,10 @@ impl<T: Send + std::fmt::Debug> MessageDispatcher<T> {
     /// A [Result] containing the [DbState::error] if it was already set, or the result
     /// of [MessageDispatcher::handle_result] otherwise.
     pub(crate) async fn run(&mut self) -> Result<(), SlateDBError> {
-        // TODO: handle panic here (similar to spawn_bg_task)
-        let result = self.run_loop().await;
+        let result = AssertUnwindSafe(self.run_loop())
```

Comment:
Ended up implementing the panic handling because some tests were hanging otherwise. They were using failpoints to trigger panics, which were resulting in hanging tests.